gradle-plugin/src/main/java/io/fairyproject/gradle/util/MavenUtil.java

```diff
@@ -35,55 +99,124 @@ public String getLatest(String module) throws IOException {
         return latest;
     }
 
-    public String getLatest(String group, String artifact) throws IOException {
-        final URL url = new URL(VERSION_URL
-                .replace("<module>", artifact)
-                .replace("<group>", group)
-        );
-        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
-        connection.setDoInput(true);
-        connection.setRequestMethod("GET");
-        connection.addRequestProperty("User-Agent", "Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:56.0) Gecko/20100101 Firefox/56.0");
-        connection.setConnectTimeout(10000);
-        connection.setReadTimeout(10000);
-
-        final int responseCode = connection.getResponseCode();
-        if (responseCode >= 200 && responseCode < 300) {
-            try (InputStream in = connection.getInputStream()) {
-                JsonObject jsonObject = new Gson().fromJson(new InputStreamReader(in), JsonObject.class);
-                for (JsonElement items : jsonObject.getAsJsonArray("items")) {
-                    final JsonObject itemObject = items.getAsJsonObject();
-
-                    return itemObject.getAsJsonObject("maven2").get("version").getAsString();
+    public String getLatest(String group, String artifact) {
+        return cacheable("getLatest", group + "|" + artifact, String.class, () -> {
+            final URL url = new URL(VERSION_URL + group.replace(".", "/") + "/" + artifact);
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            connection.setDoInput(true);
+            connection.setRequestMethod("GET");
+            connection.addRequestProperty("User-Agent", "Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:56.0) Gecko/20100101 Firefox/56.0");
```

Comment:
User agent should be a custom one if you want to do analytics tracking

Reply:
done